WooCommerce/Classes/ViewRelated/Orders/Order Creation/PaymentSection/ShippingLineDetailsViewModel.swift

```diff
@@ -0,0 +1,107 @@
+import SwiftUI
+import struct Yosemite.ShippingLine
+
+class ShippingLineDetailsViewModel: ObservableObject {
+
+    /// Closure to be invoked when the shipping line is updated.
+    ///
+    var didSelectSave: ((ShippingLine?) -> Void)
+
+    /// Helper to format price field input.
+    ///
+    private let priceFieldFormatter: PriceFieldFormatter
+
+    /// Formatted amount to display. When empty displays a placeholder value.
+    ///
+    var formattedAmount: String {
+        priceFieldFormatter.formattedAmount
+    }
+
+    /// Stores the amount(unformatted) entered by the merchant.
+    ///
+    @Published var amount: String = "" {
+        didSet {
+            guard amount != oldValue else { return }
+            amount = priceFieldFormatter.formatAmount(amount)
+        }
+    }
+
+    /// Defines the amount text color.
+    ///
+    var amountTextColor: UIColor {
+        amount.isEmpty ? .textSubtle : .text
```

Comment:
Using `.textSubtle` for the amount placeholder makes it a slightly different color from the other placeholder text in this view. It looks like this will make it match:

```suggestion
        amount.isEmpty ? .textPlaceholder : .text
```

Before|After
-|-
![Simulator Screen Shot - iPhone 13 Pro - 2022-02-18 at 12 27 57](https://user-images.githubusercontent.com/8658164/154707497-69085172-97e5-4c50-9259-a88b4ec63dda.png)|![Simulator Screen Shot - iPhone 13 Pro - 2022-02-18 at 15 01 16](https://user-images.githubusercontent.com/8658164/154707504-bd0b9bab-d524-4c17-b6bc-844b9ffb481a.png)

Reply:
Good catch! Updated in 127ea7c.